src/v0.rs

```diff
@@ -428,6 +446,8 @@ impl<'s> Parser<'s> {
     }
 
     fn skip_path(&mut self) -> Result<(), Invalid> {
+        self.push_depth()?;
```

Comment:
Why do all of these calls not have matching `self.pop_depth()` calls? This seems like it's "leaking" recursion levels, and will terminate much sooner than necessary.

Reply:
Just confirmed that there are legitimate `v0` symbols in the "1 million symbols" dataset I used for https://github.com/rust-lang/rust/pull/57967, that fail to demangle because of this bug. I'll open a fix PR for it.